features/events/api/src/main/java/org/opennms/netmgt/xml/event/Event.java

```diff
@@ -1883,7 +1883,7 @@ public String toStringSimple() {
 		if (_dbid   != null) builder.append("dbid", _dbid);
 		if (_source != null) builder.append("source", _source);
 		if (_nodeid != null) builder.append("nodeid", _nodeid);
-		if (_parms  != null) builder.append("parms", _parms);
+        if (_parms  != null) builder.append("parms",  new ArrayList<>(_parms));
```

Comment:
Do you have a copy of the stack trace for the `ConcurrentModificationException` that this resolves?

Reply:
I don't, anymore. It was on paste.opennms.eu, but is no more.

The engine reload integration tests work; I'm getting an install set up to see if it works "live".
The engines at work may not be the best test case for this, since they're not all refactored to use CEP yet.

Reply:
For example, in NMS-9907, I reported that persistence doesn't work with 6.5.0, either. But I'm not 100% confident that it isn't my rule-sets to blame.

Reply:
Can we add a comment here that explains why we're copying the `_params` array? (since it's not obvious)

Reply:
A comment that explains the `new ArrayList<>(_parms)`, or one that explains why it's even being copied into the `ToStringBuilder` object in the first place?